core/src/main/kotlin/io/saagie/updatarium/dsl/ChangelogExecutionState.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2019-2020 Pierre Leresteux.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.saagie.updatarium.dsl
+
+import io.saagie.updatarium.dsl.UpdatariumError.ChangesetError
+
+internal data class ChangelogExecutionState(val exceptions: List<ChangesetError> = emptyList()) {
+    private val hasError: Boolean = exceptions.isNotEmpty()
+
+    val report: ChangelogReport by lazy {
+        ChangelogReport(exceptions)
+    }
+
+    fun execute(failFast: Boolean, block: () -> List<ChangesetError>): ChangelogExecutionState =
```

Comment:
Maybe this function can have a unit test, because it's the new failfast mecanism.